Added initial common variables

irinazoccolini opened the pull request:
This PR adds initial common variables in `common.py` that are used for analysis purposes. 

More specifically the following variables are added:

- `FEMALE`: a `Gender` instance for female pulled from the database of initial values
- `MALE`: a `Gender` instance for male pulled from the database of initial values
- `NONBINARY`: a `Gender` instance for nonbinary pulled from the database of initial values
- `BINARY_GROUP`: a Python list of female and male `Gender` instances
- `TRINARY_GROUP`: a Python list of female, male, and nonbinary `Gender` instances
- `SWORDS_ENG`: a Python list of English stopwords from the nltk package
- `NLTK_TAGS`: a Python dictionary mapping parts of speech tags to their definitions
- `NLTK_TAGS_ADJECTIVES`: a Python list of adjective tags ('JJ', 'JJR', 'JJS')

irinazoccolini committed:
Added initial common variables

Co-Authored-By: Ashar Farooq <ashar744@gmail.com>

MBJean commented on backend/app/common.py:
```diff
@@ -1,8 +1,55 @@
+NONBINARY = Gender.objects.all().filter(label="Neo")
+
+BINARY_GROUP = [FEMALE, MALE]
+TRINARY_GROUP = [FEMALE, MALE, NONBINARY]
```
After thinking about this a bit, I think we should remove lines 8 through 13 from this file and instead supply functions that perform the database lookup. That is, something like:
```
def get_gender_female():
  return Gender.objects.all().filter(label="Female")
```

That way we can ensure that the `Gender` model lookup only occurs when needed.

If we did that, we'd presumably then simply remove the `BINARY_GROUP` and `TRINARY_GROUP` constants. 

And thinking through this further, we may not need any of these `Gender` constants at all. We'll ultimately have a user interface, and we can enforce in the interface that the user must select which `Gender`s they're going to use in their analysis. That is to say: in the original project, the analyses defaulted to the binary gender. In this project, we can more carefully ensure that the user must select which genders are of relevance and not have a default.

irinazoccolini replied:
I think this makes sense. I've gone back and removed lines 8 through 13.

MBJean reviewed (changes requested):
Great work with this! I've left a comment above that I'd be interested in your feedback on.

irinazoccolini committed:
Removed gender constants

Removed the MALE, FEMALE, NONBINARY, TRINARY_GROUP, and BINARY_GROUP constants.

farooqashar committed:
added functions for getting basic genders

farooqashar committed:
eliminated all Gender constants

MBJean reviewed (approved):
LGTM! Thanks for your work on this. Feel free to merge this in to the main branch by selecting "Merge pull request" below and give an announce in the Slack room.
